Zend/zend_ast.c

```diff
@@ -1109,6 +1117,13 @@ static void* ZEND_FASTCALL zend_ast_tree_copy(zend_ast *ast, void *buf)
 		ZVAL_STR_COPY(&new->val, zend_ast_get_constant_name(ast));
 		Z_LINENO(new->val) = zend_ast_get_lineno(ast);
 		buf = (void*)((char*)buf + sizeof(zend_ast_zval));
+	} else if (ast->kind == ZEND_AST_OP_ARRAY) {
```

Comment:
I believe Dmitry already asked you to move this down. Lists are very common, so we should likely move this down even further.

Reply:
> I believe Dmitry already asked you to move this down.

Initially it was in the top-most position. When moving it down, I moved it to the last position that does not perform a function call, though I'm seeing now it's force-inlined.

Shall I move this right above the `else`? Would it make sense to switch to a `switch()` instead?

Reply:
> Shall I move this right above the else?

I think that would be best.

> Would it make sense to switch to a switch() instead?

Would be fine too. FWIU gcc has been able to create jumptables for `if`/`else` chains for a while. Honestly, I have not checked whether that is already the case.

Reply:
> I think that would be best.

Okay. Moved it down. The switch to switch would result in a larger diff and would be best to do in a follow-up.